### PR TITLE
Remove explicit default org ID use

### PR DESCRIPTION
**Pull request by desa**
Encodes DefaultOg logic into the OrganizationsStore interface.

Prevents user from removing default organization.

Connect #2202 
Connect #2209

### CODE REVIEW

**Comment by jaredscheib** [comment]
initial comment: in the tests in server/me_test and server/auth_test, it's confusing to see the Default organization have a name other than Default and pass

File: bolt/organizations_test.go [hunk 1]
```diff
@@ -462,3 +463,70 @@ func TestOrganizationsStore_Add(t *testing.T) {
 		}
 	}
 }
+
+func TestOrganizationsStore_DefaultOrganization(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (desa) [reply bolt/organizations_test.go]: Totally thought I added one previously 👍

**Comment by desa** [comment]
@goller @jaredscheib 

I believe I addressed all the issues.

**Comment by desa** [comment]
@jaredscheib cool if I merge this? Just waiting for your approval.

**Comment by desa** [comment]
@jaredscheib merged aggressively. If anything else needs to change I'll address it in a separate PR.

**Comment by jaredscheib** [comment]
Sounds good. Only question is why we return 'unknownErrorWithMessage' consistently in the case that DefaultOrg store returns an error, instead of more explicitly what the issue is? Is that consistent with errors returned elsewhere when store accessors fail?

And I understand correctly, if ever a request does not contain an org in context, the Default will be used, yeah? Would we rather return an error if an org is missing? Or do we do this as well?

And, this slows for renaming of Default org. Seems ok to me but we may revisit this in the future.

Code looks solid. Nice work.

**Pull request merged**